healthcheck/health_check.go

```diff
@@ -0,0 +1,83 @@
+package HealthCheck
+
+import (
+	"context"
+	"time"
+)
+
+type Checker interface {
```

Comment:
Not convinced this should be an interface as initially planned.
The App using it would need to create a new struct which implemented that single function for each check needed. E.g. implementation below
```GO
type someChecker struct {
}

func (sc someChecker) CheckAppHealth(ctx *context.Context) (check *Check, err error) {
	return
}
```

An alternative is maybe to use a type instead? e.g
```GO
type CheckAppHealth func(*context.Context) (*Check, error)

var Checker CheckAppHealth = func(ctx *context.Context) (check *Check, err error){
 return
}
```

Reply:
Just spoke to @daiLlew too think he agrees on this, will leave it open for discussion whilst I finish writting the tests, but if no further feedback I will opt for using a type instead.